virtool/samples/data.py

```diff
@@ -159,6 +160,210 @@ async def find(
+                                }
+                            },
+                            {
+                                "$unwind": {
+                                    "path": "$job.user",
+                                    "preserveNullAndEmptyArrays": True,
+                                }
+                            },
+                            {
+                                "$set": {
+                                    "last_status": {"$last": "$job.status"},
+                                    "first_status": {"$first": "$job.status"},
+                                }
+                            },
+                            {
+                                "$set": {
+                                    "job.created_at": "$first_status.timestamp",
+                                    "job.progress": "$last_status.progress",
+                                    "job.state": "$last_status.state",
+                                    "job.stage": "$last_status.stage",
+                                }
+                            },
+                            {
+                                "$set": {
+                                    "job.id": "$job._id",
+                                }
+                            },
+                            {"$sort": sort},
+                            {"$skip": skip_count},
+                            {"$limit": per_page},
+                        ],
+                    }
+                },
+                {
+                    "$project": {
+                        "data": {
+                            "_id": True,
+                            "workflow": True,
+                            "created_at": True,
+                            "index": True,
+                            "job._id": True,
+                            "job.acquired": True,
+                            "job.workflow": True,
+                            "job.args": True,
+                            "job.rights": True,
+                            "job.state": True,
+                            "job.status": True,
+                            "job.user._id": True,
+                            "job.user.handle": True,
+                            "job.user.administrator": True,
+                            "job.ping": True,
+                            "job.created_at": True,
+                            "job.progress": True,
+                            "job.stage": True,
+                            "job.id": True,
+                            "job.archived": True,
+                            "ready": True,
+                            "reference": True,
+                            "sample": True,
+                            "subtractions": True,
+                            "updated_at": True,
+                            "user._id": True,
+                            "user.handle": True,
+                            "user.administrator": True,
```

Comment:
Is there no way to simplify this. Can you not just do `"job": True` to cover all job fields?

Reply:
I originally wanted to remove the "key" field from being returned in jobs, but I realised it was easier to do with an $unset. I also did this in analyses/data.